fix latest pip egg deprecation

ocefpaf opened the pull request:
Should fix #1406

ocefpaf commented on .github/workflows/build_latest.yml:
```diff
@@ -63,13 +63,14 @@ jobs:
         export PATH=${NETCDF_DIR}/bin:${PATH} 
         export NETCDF_PLUGIN_DIR=${{ github.workspace }}/netcdf-c-${NETCDF_VERSION}/plugins/plugindir
-        python setup.py install
+        python -m pip install .
```
@jswhit this is the only part that we need to fix, but I confess I got lost with the parallel errors here. Any advice?

jswhit replied:
I think this is due to the fact that mpi4py is required by setup.py to find the right includes.  Adding mpi4py to 'requires' in pyproject.py fixes it for me - but I don't think that's the right solution.  Is there a way to specify an optional build-system dependency?

jswhit commented:
@ocefpaf do you mean this [one](https://github.com/Unidata/netcdf4-python/actions/runs/14711548396/job/41284945149)?

ocefpaf commented:
> adding --no-build-isolation to pip install seems to fix it. The test using netcdf-c github master still fails, but that's a different issue.

I wonder if that is the best solution. I'm OK doing that when mixing with conda, but there must be a better way for Python packaging.

PS: I'll remove the workaround I added if this passes, just want to make one change at a time.

ocefpaf commented on setup.py:
```diff
@@ -7,6 +7,15 @@
 
+
+
+has_mpi4py = False
```
I don't like this logic. Ideally we should invert this. If netcdf-c has parallel support, we recommend users to install `mpi4py` and we do so in the tests. What do you think?

jswhit replied:
I agree there's no need to check for mpi4py if netcdf-c doesn't have parallel support - so it is a bit backwards.

ocefpaf replied:
Done.

ocefpaf commented:
This one should be ready. More changes than I would like, but the main ones are:

1. removal of Python 3.8 support
2. added Python 3.13 as the "latest" the wheel test matrix
3. added `mpi4py` as an optional dependency
